lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -229,18 +230,18 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
     this.log.debug(`${this.url}: Evaluate healthiness for unhealthy provider...`)
     this.evaluatingHealthiness = true
     try {
-      await this.getBlockNumber(CallType.HEALTH_CHECK)
+      await this.getBlockNumber_EvaluateHealthiness()
```

Comment:
I'm confused on why you need to introduce another layer, meanwhile the overriden RPC methods like `getBlockNumber` can accept the optional param `CallType` to distinguish between normal check vs others?

Reply:
I didn't add the metric for log DB sync. I realized that I made more DB sync than necessary is based on spotting the error of optimistic write failure, which indicates two write-after-read operations happens too close together. I just updated the PR to add DB sync metric reporting.